cornice/service.py

```diff
@@ -465,9 +465,12 @@ def wrapper(request):
         ob = None
         view_ = view
         if 'klass' in args and not callable(view):
+            # XXX: given that request.context exists and root-factory
```

Comment:
BTW, please take a look at the "outdated" comments I made.

Reply:
I ran into issues with the `__init__(self, request)`:

https://github.com/Cornices/cornice/blob/76b0390a333d59b0977646aacb76809c7829d3ad/tests/test_service.py#L506-#L530

These tests assert `context`.

The problem stems from `DummyAPI` trying to capture `context` when `__init__()` is being called.

https://github.com/Cornices/cornice/blob/76b0390a333d59b0977646aacb76809c7829d3ad/tests/test_service.py#L26-#L37

Naively I thought that `request.context` could be used instead. It **cannot** because the *context* per se is being *constructed* since we are in the `__init__` of the `factory`.

> No matter which view calling convention is used, the view code always has access to the context via request.context.

See https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/views.html#alternate-view-callable-argument-calling-conventions

```python
@resource(collection_path='/pets', path='/pets/{id}')
class DummyAPI(object):
    last_request = None
    last_context = None


    def __init__(self, request, context=None):
        DummyAPI.last_request = request
        DummyAPI.last_context = request.context ### Naive attempt


    def collection_get(self):
        return ['douggy', 'rusty']
```

Reply:
I don't think we want to remove the 2 failing assertions dealing with `DummyAPI.last_context`.

Reply:
From what I see in the tests and code, it looks like you managed to do it, didn't you?

Reply:
No. I couldn't because of the `DummyAPI.last_context` test-cases I cited 😢 

I didn't want to remove these tests either because they are verifying what the pyramid `context` is in each case.  See [pyramid context](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.context)

Ideally, I would have liked to rewrite the tests and assert that the `context` is an instance created by the *auto-wired* `factory` but I didn't come up with a clever solution to check what the `context` was in each case (as the tests do) given that the `context` is itself when the `__init__` method is called.

I didn't want to remove the `# XXX:` I added in case I or anyone else feels like tacking this *naunce*.